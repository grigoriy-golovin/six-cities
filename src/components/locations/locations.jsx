import React from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "./../../reducer.js";


const Locations = (props) => {
  const {offers, onChangeSity} = props;
  const sityArr = Array.from(new Set(offers.map((item) => item.city.name)));
  return <section className="locations container">
    <ul className="locations__list tabs__list">
      {sityArr.map((item) => {
        return <li className="locations__item" key={item}>
          <a className="locations__item-link tabs__item" href="#" onClick={(evt) => onChangeSity(item, evt)}>
            <span>{item}</span>
          </a>
        </li>;
      })}
    </ul>
  </section>;
};

Locations.propTypes = {
  offers: PropTypes.array.isRequired,
  onChangeSity: PropTypes.func
};

const mapStsteToProps = (state, ownProps) => Object.assign({}, ownProps, {
  // sity: state.sity,
  offers: state.offers
});

const mapDispatchToProps = (dispatch) => ({
  onChangeSity: (sityName, evt) => {
    evt.preventDefault();
    dispatch(ActionCreator.changeSity(sityName));

  },
});

export {Locations};
export default connect(mapStsteToProps, mapDispatchToProps)(Locations);
