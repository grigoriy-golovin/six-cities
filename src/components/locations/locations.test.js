import React from 'react';
import {Locations} from './locations.jsx';
import renderer from 'react-test-renderer';

it(`Locations коректно рендарится после перезапуска`, () => {
  const tree = renderer
    .create(<Locations
      sityArr={[``, ``]}
      onChangeSity={jest.fn()}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
}
);

