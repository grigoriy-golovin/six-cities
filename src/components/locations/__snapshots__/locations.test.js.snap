// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations коректно рендарится после перезапуска 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
        onClick={[Function]}
      >
        <span>
          
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
        onClick={[Function]}
      >
        <span>
          
        </span>
      </a>
    </li>
  </ul>
</section>
`;
